src/lib.rs

```diff
@@ -392,35 +393,82 @@ impl Check {
                                 Some(version),
                             )?;
 
-                            Ok(run_check_release(
-                                &mut config,
-                                crate_name,
-                                current_crate,
-                                baseline_crate,
-                                self.release_type,
-                            )?)
+                            Ok((
+                                crate_name.clone(),
+                                run_check_release(
+                                    &mut config,
+                                    crate_name,
+                                    current_crate,
+                                    baseline_crate,
+                                    self.release_type,
+                                )?,
+                            ))
                         }
                     })
                     .collect()
             }
         };
-        let success = all_outcomes
-            .into_iter()
-            .fold_ok(true, std::ops::BitAnd::bitand)?;
 
-        Ok(Report { success })
+        Ok(Report {
+            crate_reports: all_outcomes?,
+        })
```

Comment:
This has a subtle behavior change: the old code runs all the checks *even if some of them error out* because it merges the `Result` values late, whereas the new code relies on `.collect()` which will bail at the first `Err`.

Since the check output is user-visible, this seems net-negative to me? But I could probably be persuaded otherwise.

Reply:
Yes, you are right. I addressed this in [1445fb6](https://github.com/obi1kenobi/cargo-semver-checks/pull/398/commits/1445fb644e5f8d29ea26064d902fc3539332a05b)

Unfortunately using a BTreeMap instead of a Vec resulted in uglier code :/